src/services/middlewares/auth.js

```diff
@@ -0,0 +1,35 @@
+import { clearSession } from 'actions/auth';
+
+const authenticator = (httpClient, store) => {
+  httpClient.interceptors.request.use((config) => {
+    const { guestLocale, session } = store.getState().auth;
+
+    if (session) {
+      Object.assign(config.headers, session);
+    } else {
+      // Waiting for https://github.com/axios/axios/issues/2190
+      /* eslint-disable no-param-reassign */
+      config.params = config.params || {};
+      Object.assign(config.params, { locale: guestLocale });
+    }
+
+    return config;
+  });
+
+  httpClient.interceptors.response.use(
+    (response) => response,
+    (error) => {
+      if (!error.response) {
+        return Promise.reject({ errors: ['Connection error'] });
+      }
+
+      if (error.response.status === 401) {
+        store.dispatch(clearSession());
+      }
+
+      return Promise.reject(error.response.data);
```

Comment:
```suggestion
    ({ response }) => {
      if (!response) {
        return Promise.reject({ errors: ['Connection error'] });
      }

      if (response.status === 401) {
        store.dispatch(clearSession());
      }

      return Promise.reject(response.data);
```

Reply:
I remember this was done on purpose, as above is showing `response` also, maybe you prefer to change it anyway but i prefer to keep it the way it is to make the differece more clear.